src/providers/jspm.ts

```diff
@@ -181,8 +183,28 @@ async function lookupRange (this: Resolver, registry: string, name: string, rang
     case 200:
       return { registry, name, version: (await res.text()).trim() };
     case 404:
+      const versions = await fetchVersions(name)
+      const semverRange = new SemverRange(range || '*', unstable)
+      const version = versions.find(v => semverRange.has(v))
```

Comment:
We mustn't "find" the first match, we must match the best version to the semver. This is done via:

```suggestion
      const version = semverRange.bestMatch(versions, unstable);
```

We also need to handle the version not found case properly - the not found error is as important as the not built error.